### DISCUSSION TITLE
Does BatchNorm have a wrong moving average?

### DISCUSSION THREAD

**Discussion by Depersonalizc** [google/flax#2083] [Q&A] [answered]
The `BatchNorm` module falsely calculates the running average by a weighted sum (with the initial mean=0, var=1) for the first batch it receives. Should just be the mean/var of the batch.

**Comment by levskaya** [comment, marked as answer]
Arguably this just comes down to whether or not you wish to presume a prior for the mean and variance of the running averages.  Since BatchNorm running statistics are exponentially weighted in the expectation of calculating them over a large number of batches, the choice of a zero-mean / unity-variance prior doesn't in practice matter much at all, and keeps the implementation much simpler.
